src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java

```diff
@@ -378,40 +386,63 @@ public void onOpened()
-            final ItemIcon exampleStackDisplay = rowPane.findPaneOfTypeByID(LIST_ELEMENT_ID_REQUEST_STACK, ItemIcon.class);
-            final List<ItemStack> displayStacks = request.getDisplayStacks();
+            private List<RequestWrapper> requestWrappers = null;
 
-            if (!displayStacks.isEmpty())
+            @Override
+            public int getElementCount()
             {
-                exampleStackDisplay.setItem(displayStacks.get((lifeCount / LIFE_COUNT_DIVIDER) % displayStacks.size()));
+                requestWrappers = getOpenRequestTreeOfCitizen();
+                return requestWrappers.size();
             }
-            else
+
+            @Override
+            public void updateElement(final int index, final Pane rowPane)
             {
-                final Image logo = findPaneOfTypeByID(DELIVERY_IMAGE, Image.class);
-                logo.setVisible(true);
-                logo.setImage(request.getDisplayIcon());
-            }
+                if (index < 0 || index >= requestWrappers.size())
+                {
+                    return;
+                }
+
+                final RequestWrapper wrapper = requestWrappers.get(index);
+                final Box wrapperBox = rowPane.findPaneOfTypeByID(WINDOW_ID_REQUEST_BOX, Box.class);
+                wrapperBox.setPosition(wrapperBox.getX() + 10*wrapper.getDepth(), wrapperBox.getY());
 
-            rowPane.findPaneOfTypeByID(REQUESTER, Label.class).setLabelText(request.getRequester().getDisplayName(request.getToken()).getFormattedText());
-            rowPane.findPaneOfTypeByID(REQUEST_SHORT_DETAIL, Label.class)
-              .setLabelText(request.getShortDisplayString().getFormattedText().replace("§f", ""));
+                final IRequest<?> request = wrapper.getRequest();
+                final ItemIcon exampleStackDisplay = rowPane.findPaneOfTypeByID(LIST_ELEMENT_ID_REQUEST_STACK, ItemIcon.class);
+                final List<ItemStack> displayStacks = request.getDisplayStacks();
 
-            request.getRequestOfType(IDeliverable.class).ifPresent((IDeliverable requestRequest) -> {
-                if (!isCreative && !InventoryUtils.hasItemInItemHandler(new InvWrapper(inventory), requestRequest::matches))
+                if (!displayStacks.isEmpty())
+                {
+                    exampleStackDisplay.setItem(displayStacks.get((lifeCount / LIFE_COUNT_DIVIDER) % displayStacks.size()));
+                }
+                else
+                {
+                    final Image logo = findPaneOfTypeByID(DELIVERY_IMAGE, Image.class);
+                    logo.setVisible(true);
+                    logo.setImage(request.getDisplayIcon());
+                }
+
+                ColonyView view = ColonyManager.getColonyView(citizen.getColonyId());
+                rowPane.findPaneOfTypeByID(REQUESTER, Label.class).setLabelText(request.getRequester().getDisplayName(view.getRequestManager(), request.getToken()).getFormattedText());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 184 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)